pagefaultgames/pokerogue
# [Bug] Update Binding moves to affect Ghosts with damage

## What are the changes?
1. Fixes Damaging Trap moves to apply the damaging effect while still allowing them to switch out.

## Why am I doing these changes?
1. Fixes: https://github.com/pagefaultgames/pokerogue/issues/1915

## What did change?
1. Binding moves as listed here: https://bulbapedia.bulbagarden.net/wiki/Bound#Binding_moves can be applied to Ghost types in order to get the Damage over Time effects (e.g. 1/8 max hp at the end of the turn).

## How to test the changes?
1. Set your overrides.ts:
```
export const MOVESET_OVERRIDE: Array<Moves> = [Moves.GROWL, Moves.PROTECT];
export const OPP_MOVESET_OVERRIDE: Array<Moves> = [Moves.INFESTATION];
```
2. Get into a battle with 2 non-Ghost Pokemon and a Ghost Pokemon.
3. Use Growl and let the opponent hit you with the Binding move (Infestation) on a non-Ghost.
4. Verify with Protect that you are taking the DoT damage.
5. Try to switch out to a different Pokemon (it should not let you).
6. When the Binding move ends or you faint, switch to the Ghost Pokemon.
7. Use Growl and let the opponent hit you with the Binding move (Infestation).
8. Verify with Protect that you are taking the DoT damage. (**new**)
9. Try to switch out to a different Pokemon (it **should** let you). (**new**)

Verify this is working against an Opponent Trainer with a Ghost-type or not (unsure how to force this).

## Checklist
